### PR TITLE
Update the JS grammar from grammars-v4 and changes visitor/test to work with the new grammar.

### PR DESCRIPTION
**Pull request by faustocarva**
Update the JS grammar from grammars-v4 and make the change visitor and test to work with the new grammar

### CODE REVIEW

File: src/main/antlr4/br/unb/cic/js/miner/JavaScriptParser.g4 [hunk 1]
```diff
@@ -374,8 +417,7 @@ objectLiteral
     ;
 
 anonymousFunction
-    : functionDeclaration                                                       # FunctionDecl
```
Comment (rafaelcn) [inline src/main/antlr4/br/unb/cic/js/miner/JavaScriptParser.g4:377]: an interesting decision, why?

File: src/test/resources/examples/AsyncAwait.js [hunk 2]
```diff
@@ -3,9 +3,8 @@ async function f(){}
 class C {
     async method(){}
     static async method1(){}
-    async static #method2(){}
```
Comment (rafaelcn) [inline src/test/resources/examples/AsyncAwait.js:6]: should a function name prefixed with a `#` work anyway?

**Review by rafaelcn — COMMENTED** [review]
On the surface, it looks good to me but there are others that raised my eyebrows.

File: src/main/antlr4/br/unb/cic/js/miner/JavaScriptParser.g4 (hunk 1 above)
Comment (faustocarva) [reply src/main/antlr4/br/unb/cic/js/miner/JavaScriptParser.g4:377]: These are all updates from https://github.com/antlr/grammars-v4/blob/master/javascript/javascript/JavaScriptParser.g4 . I personally don't know what all changes are dealing with. But, there are bug reports and comments for all of them in the https://github.com/antlr/grammars-v4/ repo

File: src/test/resources/examples/AsyncAwait.js (hunk 2 above)
Comment (faustocarva) [reply src/test/resources/examples/AsyncAwait.js:6]: This one is a change from https://github.com/antlr/grammars-v4/pull/3530 . But, yeah, I think `#` before a method in a class makes it private, somehow.  https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Classes/Private_properties#private_methods

**Review by rafaelcn — APPROVED** [review]

**Comment by rafaelcn** [comment]
hey @faustocarva can you update the README.md with fixes to the list of features implemented by the grammar?

**Pull request merged**